lib/extensions/ar_slug.rb

```diff
@@ -0,0 +1,13 @@
+module ArSlug
+  extend ActiveSupport::Concern
+
+  included do
+    virtual_column :href_slug, :type => :string
+
+    def href_slug
+      return unless id
+      collection = Api::CollectionConfig.new.name_for_subklass(self.class)
+      "#{collection}/#{id}" if collection
```

Comment:
I feel like this should be a builder method in Api namespace, that we just call here.  Thinking something like...

```
# here

def href_slug
  Api.build_href_slug(self.class, id)
end

# there  Not sure the best module, but that's up to you
module Api
  def self.build_href_slug(klass, id)
    return unless id
    collection = Api::CollectionConfig.new.name_for_subklass(klass)
    "#{collection}/#{id}" if collection
  end
end
```

This way the knowledge for building slugs generically belongs with the API and the API specs, irrespective of having an AR instance (i.e. it can also be called (and tested) with only a type/id pair and not an instance).  And the knowledge of how to build it for AR instance then is just a simple call with some defaults from the instance.